test/tools/router_check/coverage.cc

```diff
@@ -0,0 +1,21 @@
+#include "test/tools/router_check/coverage.h"
+
+#include <algorithm>
+
+#include "envoy/api/v2/core/base.pb.h"
+
+namespace Envoy {
+void Coverage::markCovered(const Envoy::Router::RouteEntry& route) {
+  if (std::find(seen_routes_.begin(), seen_routes_.end(), &route) == seen_routes_.end()) {
```

Comment:
I think this is the inverse of what you want? If so I think your tests are not working. :)

Reply:
using std::unordered_set can possibly simplify it?
like this? https://github.com/jyotima/envoy/blob/7690b8ddb0927715840ffe4b0919bcfdf6174762/test/tools/router_check/router.cc#L173

Reply:
I think this is right. If find reaches the end of the already seen routes then we know we haven’t encountered this route before. If it doesn’t then we skip adding it as a duplicate.

I can also look at using an unordered set if that’s preferred.

Reply:
Oops sorry, my bad. Can you add a small comment?